Add win_mssql_query module: Module for query MSSQL databases

dlazz opened the pull request:
##### SUMMARY
This new module allows user to query MSSQL Server databases
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME
win_mssql_query

##### ANSIBLE VERSION

```
2.7
```

##### ADDITIONAL INFORMATION
Example: List all the databases on a SQL Server
```
tasks:
    - name: get databases
      win_mssql_query:
        query: "SELECT name FROM sys.databases" 
        server_instance: localhost
        server_instance_user: sa
        server_instance_password: "{{ sa }}"
      register: databases

    - debug:
        msg: "{{ item.name }}"
      loop: "{{ databases.output }}"
      when: databases.output != []
```
Output:
```

PLAY [win_mssql_query] *************************************************************************************************************************************************************************

TASK [Gathering Facts] *************************************************************************************************************************************************************************
ok: [ansible]

TASK [get databases] ***************************************************************************************************************************************************************************
changed: [ansible-test] => {"changed": true, "output": [{"name": "master"}, {"name": "tempdb"}, {"name": "model"}, {"name": "msdb"}]}

TASK [debug] ***********************************************************************************************************************************************************************************
ok: [ansible] => (item={'name': 'master'}) => {
    "msg": "master"
}
ok: [ansible] => (item={'name': 'tempdb'}) => {
    "msg": "tempdb"
}
ok: [ansible] => (item={'name': 'model'}) => {
    "msg": "model"
}
ok: [ansible] => (item={'name': 'msdb'}) => {
    "msg": "msdb"
}

PLAY RECAP *************************************************************************************************************************************************************************************
ansible               : ok=3    changed=1    unreachable=0    failed=0
```

Commit:
adds win_mssql_query module

Commit:
fixed credentials security, select, version in doc

Commit:
adds aliases

Commit:
fix test/sanity/pslint/ignore.txt

Commit:
fix ConvertTo-SecureString with empty string

dagwieers commented:
Beware that [the deadline](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/roadmap/ROADMAP_2_7.rst#release-schedule) for getting new features/modules accepted in Ansible v2.7 is nearing, it is set to either 2018-08-23 or 2018-08-30. If you are blocked, or you need feedback, please discuss on IRC channel #ansible-windows or add a comment to the [Windows Working Group meeting agenda](https://github.com/ansible/community/issues/294) to get it resolved.

jhawkesworth commented:
This would be a nice module to have, but I think you could give it a basic kind of idempotence without too much difficulty.
You could add 2 new parameters 'creates' and 'removes' which would take sql queries to run.  If the 'creates' query returns any results, then the main query will not be run (again).  In a similar way you could alternatively specify a 'removes' query which would work the opposite way: If the 'removes' query returns any results, then the main query would be run.  This would let you ensure that rows were not present in the database.

dlazz commented:
can you give me some example?

jborean93 commented on lib/ansible/modules/windows/win_mssql_query.ps1:
```diff
@@ -0,0 +1,72 @@
+        Fail-Json -obj $result -message "Error: $($_.exception.message)"
+    }
+    if ($res) {
+        Switch ($res.GetType()|Select-Object -ExpandProperty Name) {
```
You might be better off doing `if ($res -is [Array])` and handle that accordingly. Having codepaths based on the string of the type seems wrong to me.

dlazz replied:
I don't like it too, but the problem here is that `if ($res -is [Array])` returns true whether is an Object[] or a DataRow.

jborean93 replied:
That's interesting [DataRow](https://docs.microsoft.com/en-us/dotnet/api/system.data.datarow?view=netframework-4.7.2) doesn't show any indication that it is an Array. You could also try `$res.GetType().IsArray` and iterate from there.

Should we be only getting the first entry if the value is `Object[]`? It seems like if it is an array we would be returning all `DataRows` in that array, e.g. make `$result.output` a list of `DataRow` values.

dlazz replied:
Sorry, I was wrong :(
```
$res = Invoke-Sqlcmd -ServerInstance mysql -Query "Select top 2 name from sys.databases"
Is Array: $($res -is [Array])
Is Array: True
$res = Invoke-Sqlcmd -ServerInstance mysql -Query "Select top 1 name from sys.databases"
Is Array: $($res -is [Array])
Is Array: False
```
So I think that the code above could become:
```
if ($res) {
    if ($res -is [Array]){
        $Columns = $res[0].Table.Columns.Caption
    } 
    else {
        $Columns = $res.Table.Columns.Caption
    }
    $output = @($res |Select-Object -Property $Columns)
}
```

jborean93 commented on lib/ansible/modules/windows/win_mssql_query.py:
```diff
@@ -0,0 +1,97 @@
+DOCUMENTATION = r'''
+---
+module: win_mssql_query
+version_added: '2.7'
```
Bump to 2.8

jborean93 commented on lib/ansible/modules/windows/win_mssql_query.py:
```diff
@@ -0,0 +1,97 @@
+  server_instance:
+    description:
+      - Name of the sql server instance to query.
+    default: computername
```
Keep out the default, add a new entry to the description that says `- If not specified, the remote host will be used.`

jborean93 commented on lib/ansible/modules/windows/win_mssql_query.py:
```diff
@@ -0,0 +1,97 @@
+      - Name of a sql user with sufficient privileges.
+  server_instance_password:
+    description:
+      - Password for server_instance_user
```
When referencing another option, enclose that in I(), e.g. `I(server_instance_user)`

jborean93 commented on lib/ansible/modules/windows/win_mssql_query.py:
```diff
@@ -0,0 +1,97 @@
+      - database where the query has to be executed
+    default: master
+author:
+  - Daniele Lazzari
```
Please put your github user id in brackets after the name. This is used so the Ansible bot can ping you on issues and pull requests for this module.

jborean93 commented on lib/ansible/modules/windows/win_mssql_query.py:
```diff
@@ -0,0 +1,97 @@
+  - Daniele Lazzari
+notes:
+  - If the user that runs ansible has the appropriated permission on the db,
+  - you don't need to set C(server_instance_user) and C(server_instance_password).
```
No need to put this in a new list entry, just put `you` in the newline where it is but remove `-`. You may want to add in that CredSSP, Kerb with cred delgation or become must be used for implicit auth to work.

jborean93 commented on lib/ansible/modules/windows/win_mssql_query.ps1:
```diff
@@ -0,0 +1,72 @@
+#!powershell
+# -*- coding: utf-8 -*-
```
This isn't needed in PowerShell.

jborean93 commented on lib/ansible/modules/windows/win_mssql_query.ps1:
```diff
@@ -0,0 +1,72 @@
+    $output = @()
+    try {
+        if (!($CheckMode)) {
+            if (!($ServerInstanceUser) -or !($ServerInstancePassword)) {
```
Consider using splatting here instead;

```
$sql_params = @{
    ServerInstance = $ServerInstance
    Database = $Database
    Query = $Query
    QueryTimeout = $QueryTimeout
}
if ($ServerInstanceUser -and $ServerInstancePassword) {
    $SQLCredential = New-Object TypeName System.Management.Automation.PSCredential ($ServerInstanceUser, $ServerInstancePassword)
    $sql_params["Credential"] = $SQLCredential
}

Invoke-SqlCmd @sql_params
```

The benefit to this is you only call `Invoke-SqlCmd` in one spot, also it's a lot easier to deal with if statements when you aren't checking the false result.

dlazz replied:
Excellent, I didn't know it was possible, I was looking for a solution like this and I'll definitely do it.

jborean93 reviewed (commented):
Looks great, just a few comments but thanks for the work you've done so far.

Commit:
code review

Commit:
add win_mssql_query module

Commit:
archive: Fix check if archive is created in path to be removed (#29420)

Commit:
Added Ansible UTM Module for managing reverse proxy auth profile entries.

MatrixCrawler committed:
Added Ansible UTM Module for managing reverse proxy frontend entries

MatrixCrawler committed:
Added Ansible UTM Module for managing reverse proxy location entries

Commit:
- Fix sanity checks
  - Fixed documentation trailing whitespaces
  - Added missing/broken documentation

Commit:
- Fix sanity checks
  - Fixed documentation trailing whitespaces
  - Added missing/broken documentation

Commit:
- Fix sanity checks
  - Fixed documentation trailing whitespaces
  - Added missing/broken documentation

abadger committed:
Add a CLIArgs Singleton class that will hold the parse cli arguments

abadger committed:
Save the command line arguments into a global context

* Once cli args are parsed, they're constant.  So, save the parsed args
  into the global context for everyone else to use them from now on.
* Port cli scripts to use the CLIARGS in the context
* Refactor call to parse cli args into the run() method
* Fix unittests for changes to the internals of CLI arg parsing
* Port callback plugins to use context.CLIARGS
  * Got rid of the private self._options attribute
  * Use context.CLIARGS in the individual callback plugins instead.
  * Also output positional arguments in default and unixy plugins
  * Code has been simplified since we're now dealing with a dict rather
    than Optparse.Value

abadger committed:
Split up the base_parser function

The goal of breaking apart the base_parser() function is to get rid of
a bunch of conditionals and parameters in the code and, instead, make
code look like simple composition.

When splitting, a choice had to be made as to whether this would operate
by side effect (modifying a passed in parser) or side effect-free
(returning a new parser everytime).

Making a version that's side-effect-free appears to be fighting with the
optparse API (it wants to work by creating a parser object, configuring
the object, and then parsing the arguments with it) so instead, make it
clear that our helper functions are modifying the passed in parser by
(1) not returning the parser and (2) changing the function names to be
more clear that it is operating by side-effect.

Also move all of the generic optparse code, along with the argument
context classes, into a new subdirectory.

rajaspachipulusu17 committed:
Pluribus Networks admin service module with unit tests (#49854)

* Pluribus Networks admin service module with unit tests
* Documentation fix
* import module changes
* Removed unused imports
* Fix return type

rajaspachipulusu17 committed:
Pluribus Networks dhcp filter module with unit tests (#49848)

* Pluribus Networks dhcp filter module with unit tests
* Added type which was missing in doc
* Removed unwanted global variable
* Fix return type

Commit:
Added organization in the scm_credential get (#49884)

* Added organization in the scm_credential get

* Fallback looking for cred in project org

* Tests project with multi org credential

* Fixed CI issue

* Added changelog fragment

sn4kebite committed:
rabbitmq_binding: Fix using empty routing key (#48597)

* rabbitmq_binding: Fix using empty routing key

If the routing key is an empty string we need to use the special props
value ~ in the URL. Otherwise the last part of the URL will be empty,
which will instead be equivalent to the "list all bindings" API call
which always returns 200 OK (as long as the exchange and queue exists).

* rabbitmq_binding: Move routing_key test outside format

Test the routing_key value and set the props value outside the format
call. We leave the original routing_key untouched since its original
value is later needed in create().

wojtek0806 committed:
corrected valid type values in RETURN (#50148)

bcoca committed:
dont think all files are valid cause libs present (#50103)

verify_file was improperly always returning true if pyvimomi and requests libs were correct
  moved library checking to parse, avoid unneded errors unless the file is actually meant for
  this plugin

Qalthos committed:
Allow persistent connection plugins to queue messages back to ansible-connection (#49977)

* Connections can queue messages to be returned from ansible-connection

* Provide fallback for invalid display level

* Strip display from plugins

* Route messages through helper method to try to avoid improper appends

Commit:
Refactoring of cnos_interface module like what followed by other vendors (#49927)

* Refactoring of cnos_interface module like what followed by other vendors

* To remove cnos-interface from E326 validation ignore list

* Effect of default parameters impacted UT

arnested committed:
Fix lastpass lookup error

Fixes #42062.

bcoca committed:
Revert range change (#50155)

* Revert "Fix incorrect examples with random filter (#50137)"

This reverts commit 9a7dbd52139f22c9c0e5f3f127efb0f8956c6f6a.

The correction is incomplete and also based on a 'fix' that was supposed to have been reverted already

* Revert "Added `+1` to the `end` in `random` filter so that it was inclusive (#27215)"

This reverts commit ea2b89c7ae69424055652db30f49182d4a987bf7.

reverted fix as agreed at the time, but missed by maintainers.

AlanCoding committed:
update openshift inventory plugin docs (#49901)

mhoffmann1 committed:
VMware: Update example vmware_deploy_ovf (#49893)

* Update vmware_deploy_ovf.py
Provided more detailed example using additional, often used parameters
Co-Authored-By: mhoffmann1 <marcin.hoffmann@intel.com>

* Replace variable values in example task
Co-Authored-By: mhoffmann1 <marcin.hoffmann@intel.com>

mattclay committed:
Add retries for Invoke-ScriptAnalyzer in pslint.

Hopefully this will work around the intermittent CI failures due
to NullReferenceException, which then succeed on a retry.

acozine committed:
Command module docs: args vs argv (#49907)

* add note about 'args', update examples
* Update lib/ansible/modules/commands/command.py

Co-Authored-By: acozine <879121+acozine@users.noreply.github.com>

bmillemathias committed:
Fix typo in ds_type in examples section (#50166)

replace ds_type elasticisearch by elasticsearch

richlv committed:
Update zabbix_action.py (#50149)

Mostly typo/grammar fixes. "Zabbix Server" was changed to just "Zabbix", as it could be slightly confusing - some people might interpret this referencing the server process, although API matters. "capital-case" was changed, as it usually is one of "capital" or "upper-case".

+label: docsite_pr

samdoran committed:
Fix ansible-runner test

Change test to support runner_on_start in Ansible 2.8 and ansible-runner 1.2.0

Qalthos committed:
Log netconf connection a la paramiko (#50169)

Commit:
optimising vmss tests (#50175)

abadger committed:
Cleanups and fixes to cli

* Mark methods which are really functions as staticmethod
* Fix calls to other staticmethods to use the subclass rather than the
  base class so that any inheritance overriding will be honored.
* Remove unnecessary logic and dead code
* Fix a typo in a docstring of how to implement subclass init_parser()
  methods
* Call superclass's post_process_args in ansible-doc
* Fix copyright comment according to suggested practice

bincyber committed:
New Module: Add bitbucket_deploy_key (#27467)

* added new module to manage deploy keys for BitBucket repositories
* Modified module to pass CI tests
* Updated bitbucket_deploy_key module
* updated message returned when there is an existing deploy key
* bumped version_added to 2.8
* fixed issues reported by ansible-test

ckotte committed:
VMware: Improve vmware_host_ntp module (#47273)

Akasurde committed:
Fix type in RETURN for bitbucket_deploy_key

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

YouhuaLi committed:
add range_start and range_end agrs when query vm by a single id (#50056)

Commit:
fixing network client version (#50182)

Commit:
sku in the sample became invalid, so updating (#50177)

NilashishC committed:
Fix eos terminal plugin (#49920)

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

justjais committed:
New Module for creating, managing Infoblox NIOS nameserver groups (#49124)

* adding new nios_nsgroup module

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* adding new nios_nsgroup module

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* adding new nios_nsgroup module

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* fix shippable errors

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* added unit tests

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

mdrose committed:
influxdb_user: Fixed unhandled exception on invalid login credentials (#50134)

maiqueb committed:
os_security_group_rule proper module exit (#50076)

When the security group the rule belongs to does not exist and
the state is absent, the module is not properly exited, leading
to a playbook execution failure.

Fixes issue #50057

dbnicholson committed:
Skip non-management Packet IP addresses (#48168)

Non-management (elastic) IP addresses require manual configuration on
the host as described in https://help.packet.net/article/54-elastic-ips.
Skip those so that only the automatically configured management
addresses are used. Otherwise, a non-routable address may be returned in
the inventory.

warusadura committed:
Update `Unknown error` to specific error message (#50129)

* Update `Unknown error` to specific error message

outputs the Exception received rather than just 'Unknown error'
fixes: issue/49713

* Update `Unknown error` to specific error message

improves the error message

hanshell committed:
Update playbooks_intro.rst (#50187)

carchi8py committed:
Rewrite (#49570)

gundalow committed:
Remove incorect author (#50191)

Akasurde committed:
Move doc fragments to changelogs/fragments dir (#50179)

Move changelog fragment to changelogs/fragments dir

richlv committed:
Update ini.py (#50171)

Typo/grammar/consistency fixes.

jonpugh committed:
Change the Packet.net section to be more consistent with the other *Cloud Guide pages. (#50170)

<!--- Your description here -->

+label: docsite_pr

Zempashi committed:
List correctly current PV in "lvg" module: fix lvg reduce  (#49731)

* Refactor integration test for lvg module to introduce grow/reduce test

* List correctly current PV in lvg module: fix lvg reduce

Previous behaviour was to only take into account PV passed in 'pvs'
argument. This lead to reduce not working as expecting:

* with state=present and list of wanted pvs, lvg found only the pvs to
  add or already present and ignored the pv to remove (obviously absent
  from the list of given PV)

* with state=absent and a pv to remove, lvg found that the remaining pvs
  list is empty (ignoring possible other PV in the vg) and decides to
  remove the vg entirely (as supposely no PV are left anymore to store
  lvm metadata)

* Add changelog fragment

fatmcgav committed:
Jenkins_plugin: Handle 'latest' version when installing plugin for first time (#49723)

* Handle 'latest' version when installing module for first time

Otherwise, a fresh install of a plugin with 'version: latest' gets installed without its dependencies, rendering the plugin effectively useless.

* Add changelog

felixfontein committed:
Prevent error if module.params['path'] has no path included. (#50106)

dagwieers committed:
MSC: Various bugfixes and features enhancements (#50200)

* MSC: Various bugfixes and features enhancements

This PR includes:
- Lookups of roles, labels and domains
- Auto-create new labels
- Improvements to comparing complex datastructures
- Force removal of sites
- Support non top-level queries
- Document internal functions
- Add parameter types to modules
- Fix documentation examples
- Improvements to idempotency wrt. returning changed
- Support site locations
- Update permission list
- Various improvements to integration tests

* Fix Ci issues

dagwieers committed:
Docs: Show parameter types (in purple) (#49966)

* Docs: Show parameter types (in purple)

* Changes based on feedback

* Remove leftover statement after review

* Simplify TOC and support section

* Add missing 'v' to version_added

* Remove the v for version

* Update docs/templates/plugin.rst.j2

Co-Authored-By: dagwieers <dag@wieers.com>

* Update docs/templates/plugin.rst.j2

Co-Authored-By: dagwieers <dag@wieers.com>

* Move Author into Support section

* Avoid more "isn't included in any toctree" errors

* Add Red Hat support section, list module status

abadger committed:
Move the arguments module into cli/ and context_objects into utils

* Note: Python2 is not as intelligent at detecting false import loops as
  Python3.  context_objects.py cannot be added to cli/arguments because it
  would set up an import loop between cli/__init__.py,
  cli/arguments/context_objects.py, and context.py on Python2.

ci_complete

cognifloyd committed:
Make modprobe module check for builtins as well (#37150)

Without this modprobe always reports changed when modprobe-ing a builtin module.

With this, if a kernel module is a builtin, the modprobe module will:
- succeed (without incorrectly reporting changed) if ``state`` is ``present``;
- fail if ``state`` is ``absent``

The failure will have whatever error message modprobe returns when
attempting to remove a builtin module. For example:
``modprobe: ERROR: Module nfs is builtin.``

Qalthos committed:
Fix some handling of log messages from ansible-connection (#50214)

zharalim committed:
Add more detailed documentation on how to use multiple inventories (#47586)

* Add a new section on how to use multiple inventory sources w/ examples

Co-Authored-By: zharalim <zharalim@outlook.com>

Akasurde committed:
Misc documentation update (#50190)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

jborean93 committed:
Added supershipit to botmeta sanity test (#50219)

rambleraptor committed:
Giving Google engineers supershipit on GCP Ansible resources/plugins/tests (#50218)

dagwieers committed:
win_updates: Correctly report changes on success (#50188)

* win_updates: Correctly report changes on success

* Add changelog fragment

dagwieers committed:
win_firewall_rule: Remove invalid 'bypass' action (#50207)

* win_firewall_rule: Remove invalid 'bypass' action

This fixes #50143

* Add changelog fragment

plastikat committed:
Fix docs typo at Special Variables (#50230)

mattclay committed:
Add `env` command to ansible-test and run in CI. (#50176)

* Add `env` command to ansible-test and run in CI.
* Avoid unnecessary docker pull.

anasbadaha committed:
Fix Issue #49612 (#50073)

* Fix Issue #49612

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Pep8 failures

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Add validation for send empty group name

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix Error message

Signed-off-by: Anas Badaha <anasb@mellanox.com>

anasbadaha committed:
Fix Issue #39598 (#49982)

* Fix Issue #39598

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix unit test for onyx_config

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Add comments for save running config

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Enhance onyx_config bug fix

Signed-off-by: Anas Badaha <anasb@mellanox.com>

* Fix result['changed'] = True comment

Signed-off-by: Anas Badaha <anasb@mellanox.com>

Tomorrow9 committed:
VMware: add VBS configuration support for Windows 10 (#48595)

* Fix space issue
* fix whitespaces issue

ckotte committed:
Revert changes in vmware_vcenter_statistics(#49537)

Akasurde committed:
postgresql_idx: misc fixes (#50199)

* Removed unused import
* Removed 'login_password' from exit_json call
* Removed unused db, cond variable

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Correct match test documentation (#50229)

Match test matches zero or more characters at the begining of the string.

Fixes: #32366

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
fixed invalid samples (#50227)

* fixed invalid samples

Qalthos committed:
Add backup parameter to cli_config (#50206)

* Add backup parameter to cli_config

* Add a unit test for cli_config backup

Qalthos committed:
Merge backup functionality across config action plugins (#50208)

* Merge backup functionality across config action plugins

* Port updated cli_config

Qalthos committed:
Log device interaction and push labeled logs back to controller (#50028)

* Add session tracing support for network_cli, netconf and httapi connection

*  Add `persistent_log_messages` configuration option to log device inteaction
   in log file for network_cli, netconf and httapi connection
   type
*  Log jsonrpc request and response in log file is configuration option
   is enabled

* Update docs to talk about warning shown when persistent_log_messages is on

rambleraptor committed:
New GCP Module: gcp_sql_user_facts (#48273)

Qalthos committed:
Allow AnsibleVaultEncryptedUnicode to pass through exec_jsonrpc (#48306)

* Allow AnsibleVaultEncryptedUnicode to pass through exec_jsonrpc

* Add changelog

rambleraptor committed:
New GCP Module: gcp_sql_instance_facts (#48271)

rambleraptor committed:
GCP Bug Fixes (#49458)

* Bug fixes

* fixes as of 12/5/2018

* changes as of 12/06/2018

rambleraptor committed:
New GCP Module: gcp_sql_database_facts (#48269)

rambleraptor committed:
New GCP Module: gcp_spanner_instance_facts (#48267)

rambleraptor committed:
New GCP Module: gcp_spanner_database_facts (#48265)

rambleraptor committed:
New GCP Module: gcp_resourcemanager_project_facts (#48263)

rambleraptor committed:
New GCP Module: gcp_iam_service_account_facts (#48257)

rambleraptor committed:
New GCP Module: gcp_container_node_pool_facts (#48251)

rambleraptor committed:
New GCP Module: gcp_container_cluster_facts (#48249)

rambleraptor committed:
New GCP Module: gcp_compute_interconnect_attachment_facts (#48217)

gitaaron committed:
ansible-doc: update faq to include common z/OS pitfalls (#18555) (#49504)

* ansible-doc: update faq to include common z/OS pitfalls (#18555)
Co-Authored-By: gitaaron <aaron.surty@gmail.com>

dagwieers committed:
Fix typo

glitchcrab committed:
update github handle (#50235)

lingfish committed:
Update nios_aaaa_record.py (#50271)

Should be ipv6.

+label: docsite_pr

Akasurde committed:
postgresql_user: set type of conn_limit (#50178)

conn_limit type is set to 'int'. This will allow module to compare conn_limit with record value without type casting.

Fixes: #38118

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

curry9999 committed:
Add colon after module name (#50274)

* A colon is not described

* dddd

* Update aws_kms_facts.py

kvaps committed:
fix: ansible_kubectl_ssl_ca_cert variable (#50217)

* fix: mistake for ansible_kubectl_ssl_ca_cert variable

* fix: K8S_AUTH_VERIFY_SSL

felixfontein committed:
Elliptic curve tests for crypto modules (#50109)

* Add openssl_csr ECC test.

* Add openssl_publickey ECC test.

* Add openssl_certificate ECC test.

gundalow committed:
Revert TEMP HACK: Upload all to codecov (#50279)

Codecov.io have made some improvements to results processing.
Upload all the coverage data to see is codecov can process this.

Reverts https://github.com/ansible/ansible/pull/47515

ganeshrn committed:
Fix for AttributeError in network_cli (#50284)

*  Value of prompt_answer is wrongly used as `self.prompt_answer`
   while logging it in message queue

rajaspachipulusu17 committed:
Pluribus Networks igmp snooping module with Unit tests (#50146)

rubentsirunyan committed:
Zabbix Action: Idempotency fix (#50297)

NilashishC committed:
Change net_* default behaviour on missing module (#49923)

* Fix net_base default behaviour

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

* New fix

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

ushuz committed:
docker_image: add network parameter (#50313)

* docker_image: add network parameter

Add a network parameter to the docker_image module to specify the
network to use for RUN commands.

Called it network instead of network_mode as the latter seems like a
legacy of when there were just a few default options to choose from,
while now the name of an arbitrary network can be specified.

Fixes #21433

* Format description

* Update docker_image option_minimal_versions

* Ensure network_mode param optional

maxamillion committed:
Provide both service state and status when possible in service_facts (#49618)

* Combine systemd units/unit-files output for service_facts

Fixes #47118

Previously we were only taking the output of `systemd units` which
would leave out information about the service units that were
disabled, static, masked, etc. Now we're aggregating the results so
that anything not active/inactive/dead at least is pulled as fact
data with it's state provided.

Signed-off-by: Adam Miller <admiller@redhat.com>

* provide state and status information about services

Fixes #45730

Signed-off-by: Adam Miller <admiller@redhat.com>

felixfontein committed:
ACME: add seealso references (#50320)

* Add seealso references to ACME modules.

* Bump to latest drafts.

felixfontein committed:
Improve tagging examples. (#50321)

sdodsley committed:
Fix bad placement of array variable (#50256)

curry9999 committed:
Documentation remove escape r (#50277)

* A colon is not described

* dddd

* Update branch

gundalow committed:
Debug: Size of coverage file (#50380)

felixfontein committed:
ufw: add integration tests (#50374)

* Add basic integration tests for ufw.

* Being more honest.

dagwieers committed:
Add missing aliases file (#50357)

gundalow committed:
Disable mysql_db (#50385)

Akasurde committed:
Temporarily disable mysql_user testcases (#50395)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Temporarily disable mysql_variables (#50403)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

sdodsley committed:
Pure Storage facts fix (#50349)

* Add module for Pure Storage FlashBlade to manage directory services

* Fix facts not correctly passing into ansible_facts dict

abadger committed:
Cleanups to the common.sys_info API

* Move get_all_subclasses out of sys_info as it is unrelated to system
  information.
* get_all_subclasses now returns a set() instead of a list.
* Don't port get_platform to sys_info as it is deprecated.  Code using
  the common API should just use platform.system() directly.
* Rename load_platform_subclass() to get_platform_subclass and do not
  instantiate the rturned class.
* Test the compat shims in module_utils/basic.py separately from the new
  API in module_utils/common/sys_info.py and module_utils/common/_utils.py

resmo committed:
vultr_server: fix multiple ssh keys not handled (#50337)

* vultr_server: fix multiple ssh keys not handled

* add changelog fragment

felixfontein committed:
docker_secret, docker_config: allow to base64-decode data (#49688)

* Adjust docker_config and docker_secret to be more similar.

* Add data_is_b64 parameter to docker_secret and docker_config.

* Add changelog.

gundalow committed:
Disable zabbix_host 50386 (#50427)

yuwzho committed:
rename force to force_delete_nonempty (#50290)

rverchere committed:
Add support for NRDP notifications (#39297)

* Add support for NRDP notifications

* Correct pep-8 check, adding copyright

* Correct pep-8 check using autopep8

* Update nrdp to 2.6 version

* Update nrdp to 2.7 version

* Use internal ansible url and options modules

* Remove useless comments

* Add option to validate https certs

* Update nrdp to 2.8 version

* Correct nrdp pep8

abadger committed:
Fix for the new 2.8 feature to have modprobe check for builtins.

uname output includes a newline.  strip() the newline so that the
filepath we construct points to the right place.

ericzolf committed:
Fix firewalld module failing on missing protocol. (#50242)

Under Python 3.7 at least, the split of the port field fails
ungracefully if there is no slash. The fix also addresses the
case of an empty protocol after the slash.

acozine committed:
uses correct highlighting (#50457)

wojtek0806 committed:
fixed a bug in transform_name (#50213)

wojtek0806 committed:
added new module to manipulate signatre sets on ASM policy (#50209)

felixfontein committed:
Sort suboptions and subresults in docs. (#50315)

Fixes #50041.

maxamillion committed:
fix order of dnf api operations so transactions don't fail (#50038)

* fix order of dnf api operations so transactions don't fail

Previously dnf.base.fill_sack() was called before
dnf.base.update_cache() which apparently breaks dnf transaction
logic as per https://bugzilla.redhat.com/show_bug.cgi?id=1658694

Fixes #49060

Signed-off-by: Adam Miller <admiller@redhat.com>

* add changelog and test case

Signed-off-by: Adam Miller <admiller@redhat.com>

maxamillion committed:
add changelog for #50242 (#50480)

Signed-off-by: Adam Miller <admiller@redhat.com>

jbenden committed:
Add support for Kali Linux detection. (#50331)

Signed-off-by: Joseph Benden <joe@benden.us>

smueller18 committed:
disable docker_network ipam tests (#50477)

felixfontein committed:
docker_swarm_service: Fix typo for option type (#50483)

* Fix typo for option type in docker_swarm_service.

* Add changelog.

felixfontein committed:
openssl_csr: idempotency doesn't work correctly for keyUsage (#50361)

* Fix key usage idempotency bug.

* Extend tests.

* Add changelog.

felixfontein committed:
openssl_*: prevent error when path includes no path (#50322)

* Prevent error when path includes no path.

* Add changelog.

lolcube committed:
add openssh_cert module (#49605)

* add openssh_cert module

* fix some linter issues

* address review feedback

* add basis for integration tests

* change valid datetime formats

* handle principals as set not as list

* handle principals as set not as list

* handle principals as set not as list

it-praktyk committed:
Add choices of the state parameter documentation (#50481)

* Add choices of the state parameter documentation

* Consistency

machacekondra committed:
ovirt_host: Improve failed upgrade check (#50489)

fretis committed:
Update playbooks_loops.rst (#50437)

Error in the example for exclusion pattern

+label: docsite_pr

Akasurde committed:
Update changed_when and failed_when examples (#50411)

Added examples in playbooks_error_handling doc for handlining
multiple conditions in changed_when and failed_when

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
gcp: documentation update (#50183)

* gcp: documentation update

* Update example about dynamic inventory
* minor typo fixes in gcp_utils
* Additional information about enabling inventory plugin in ansible.cfg

partially fixes: #44404

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

richlv committed:
Update zabbix_action.py (#50490)

* Add trailing dots for consistency.
* Document how the current host can be specified for "run_on_hosts".

+label: docsite_pr

Commit:
Fixes #37209 - Added complex return value, instance (#49989)

felixfontein committed:
Docs: add Pygments lexer for Ansible output (#50318)

* Add specialized lexer for Ansible output.

* Make linter happy.

* Use different tokens.

agowa338 committed:
Make WinRM security warning more explicit (#50263)

Make WinRM security warning more explicit.
Currently the warning is "viewed by anyone", this is a much lower risk than arbitrary command injection. Therefore the risk should be phrased appropriately.

+label: docsite_pr

dagwieers committed:
Windows: Add missing parameter types and doc fixes (#50232)

* Windows: Add missing parameter types and doc fixes

This PR includes:
- Adding missing parameter types
- Various documentation fixes

* Update lib/ansible/modules/windows/win_copy.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Update lib/ansible/modules/windows/win_credential.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Update lib/ansible/modules/windows/win_domain_computer.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Update lib/ansible/modules/windows/win_domain_user.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Update lib/ansible/modules/windows/win_environment.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Update lib/ansible/modules/windows/win_psexec.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Update lib/ansible/modules/windows/win_uri.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Update lib/ansible/modules/windows/win_wait_for.py

Co-Authored-By: dagwieers <dag@wieers.com>

* Ensure docstrings are raw strings

drewmullen committed:
allow for vault enterprise namespaces (#50462)

* enable namespaces feature for hashi_vault lookup

* include version_added dict in options documentation

jdorel committed:
Add galaxy.html#dependencies link (#49984)

* Add galaxy.html#dependencies link

+label: docsite_pr

s-hertel committed:
Add unstable alias to s3_bucket integration tests (#50508)

jborean93 committed:
dnf - fix conf_file loading (#50515)

jborean93 committed:
ansible test: re-enable mysql tests with fix (#50430)

sivel committed:
Give IncludedFile more context via ansible_search_path (#50045)

* Give IncludedFile more context via ansible_search_path to template lookups. Fixes #49969

* Update units

dagwieers committed:
urldecode filter for Jinja2 (#28503)

* urldecode filter for Jinja2

We needed this in order to deconstruct correct URLs using Jinja2.
And we might as well upstream this.

* Add integration tests

* Fixes for Python 3

* Add urlencode for older Jinja2

dagwieers committed:
ACI: Implement encoded query_string (#50358)

This requires urldecode support in Ansible

jborean93 committed:
ansible-test: docker_network set as untable (#50526)

* Revert "disable docker_network ipam tests (#50477)"

This reverts commit 4dd2b8e2d004418a981ff7c9552792d44f19131a.

* test - docker_network mark as unstable

richlv committed:
Update documentation for mysql_db notes (#50444)

* Simplify the requirement section to avoid duplicating info already in mysql_document_fragment.
* Package installation section simplified, mentioned RHEL; a typo fix in "than".
* added dnf for Fedora

Akasurde committed:
Handle exception raised in recursive_finder API (#49590)

User module can contain Indentation errors or syntax errors.
Handle AST exceptions rather than showing traceback while importing such module.

Fixes: #21707

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

justjais committed:
Added new module to support NIOS Fixedaddress DHCP property (#49119)

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* nios fixedaddr new module

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* updating name

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* added unit tests

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

Commit:
remove forced lowercase on group_name (#50445)

yungezz committed:
azure_rm_webapp: fix updating app settings (#49181)

unrinfosec committed:
Correct Errors in Documentation (#50507)

* Correct Errors in Documentation

According to the module, "compose" is not required, but the docs said it was - removed the "required" tag. Additionally, the example for removing a stack did not include the required "name" option (this method is why "compose" is not required.
+label: docsite_pr

* Removed change to description

Corrected issue to my "correction" on the description.

* Removed required false

* Removed all instances of required: false.

ganeshrn committed:
Fix backup issue in network config modules and network action plugins common code refactor (#50301)

* Fix backup issue in network config modules

*  Fix `get_working_path` not found issue introduced due to
   backup config code refactor (PR #50208)

*  Further refactor config related action plugins to minimize
   duplicate code

*  Remove unwated imports in config action plugins

* Add common network class for action plugin and related code refactor

* Fix review comment

trishnaguha committed:
add privileged role validation for nxos become (#50312)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

ganeshrn committed:
Fix junos terminal error regex (#50538)

*  Change junos terminal error regex to read error string
   correctly.

jborean93 committed:
dnf/yum - added install_weak_deps option (#50525)

* dnf/yum - added install_weak_deps option

* skip creating weak dep packages if rpm doesn't support it

* fix rpm check for older hosts

abadger committed:
Add a porting guide entry for ansible_distribution facts (#50251)

* Add a porting guide entry for ansible_distribution facts

Switching away from platform.distro() will cause changes sometimes due
to the new code using new sources of information that may be out of sync
with the old ones.  Just have to make people aware of that and also what
we are doing to mitigate it when appropriate.

* wordsmithed, added links for new distro backend

wojtek0806 committed:
adds apm policy fetch module (#50534)

wojtek0806 committed:
Changes exists() method logic in asm policy fetch, corrects documentation (#50560)

wojtek0806 committed:
adds bigip apm policy import (#50559)

wojtek0806 committed:
Corrects doc strings in bigip_asm_policy_* modules (#50561)

sivel committed:
Add new AnsibleTemplateError to more easily catch templating issues (#50563)

* Add new AnsibleTemplateError to more easily catch templating issues. Fixes #50154

* Add changelog fragment

sivel committed:
module_utils/urls.py is core (#50567)

* module_utils/urls.py is core

* More core supported module_utils

* More missed core module_utils

kairoaraujo committed:
new module: Configure LVM and NFS file systems for AIX (#30810)

* new module: AIX LVM file system and NFS

This module creates, removes, mount and unmount LVM and NFS file system
for AIX using /etc/filesystems. For LVM file systems is also possible
to resize the file system.

* better parameters options structure

better parameters options structure

* Improved file system resize returns

Added better tratment for return codes for file system resize.
When a resize is not possible because no enough space on lv or
shrink is not allowed.

* improved doc and creation file system return code

- improved doc
- creation file system return code 10 was treated.

* Doc recomendations, dict result, line limit

- Added doc recomendations
- Changed return to dict results on main()
- Using 159 columns for code limit

* wrong changed return when file system is already

Fixed wrong changed return when file system is already mounted.
When the file system is already mounted the return for changed
is False.

* Fixed description and included playbook for tests

- Fixed description
- Included playbook for manual tests

* Various small bits to get this merged ASAP

* Rename test/legacy/aix_filesystem.yml to test/integration/targets/aix_filesystem/tasks/main.yml

Move integration test to its proper location

* Create aliases

* Fix CI issues

richlv committed:
Improve style and fix typos in the Zabbix module titles and descriptions (#50575)

spoyd committed:
(action/script) Fix windows absolute path detection when executing powershell script (#50365)

rajaspachipulusu17 committed:
Pluribus Networks module for connection stat settings with UT's (#50186)

* Pluribus Networks module for connection stat settings

Commit:
adding module azure_rm_virtualmachinescalesetinstance_facts (#50440)

mkrizek committed:
Fix searchpath in the template lookup (#50488)

rcarrillocruz committed:
Checkpoint httpapi plugin (#49929)

* Add checkpoint httpapi plugin and access rule facts module

* WIP checkpoint_access_rule module

* Add publish and install policy, plus fix empty json object request for publish

* Refactor publish and install_policy onto module_utils

* Add update resource logic

* Add checkpoint_host_facts module

* Return code and response on get_acess_rule function

* Add checkpoint_host module

* Add checkpoint_run_script module

* Add checkpoint_task_facts module

* Show all tasks if no task id is passed

Note, this is only available on v1.3 of Checkpoint WS API

* Add update logic to checkpoint host

* Add full details on get task call

* Add checkpoint httpapi plugin

* Fix pep8

* Use auth instead of sid property and return False on handle_httperror method

* Fix version in docstring

* Remove constructor

* Remove Accept from base headers

* Do not override http error handler and assign Checkpoint sid to connection _auth

There is scaffolding in the base class to autoappend the token, given
it is assigned to connection _send

* Use new connection queue message method instead of display

* Remove unused display

* Catch ValueError, since it's a parent of JSONDecodeError

* Make static methods that are not used outside the class regular methods

* Add missing self to previously static methods

* Fix logout

Was carrying copy pasta from ftd plugin

* Remove send_auth_request

* Use BASE_HEADERS constant

* Simplify copyright header on httpapi plugin

* Remove access rule module

* Remove unused imports

* Add unit test

* Fix pep8

* Add test

* Add test

* Fix pep8

mkrizek committed:
Allow for vaulted templates in template lookup (#49819)

Fixes #34209

dagwieers committed:
raw is transport-agnostic

+label: docsite_pr

abadger committed:
Clarify the change made to reverse_inventory

Qalthos committed:
Copy detailed logging to persistent.py so local can benefit, too. (#50614)

dagwieers committed:
xml: Improve documentation and examples (#50602)

This PR includes:
- Type information for parameters
- Added 'seealso' section to documentation
- Improve examples

devjatkin committed:
add env variable VAULT_AUTH_METHOD (#50572)

resmo committed:
tests: vultr: convert legacy tests to targets  (#50156)

mattclay committed:
Fix run.py --branch argument.

The API docs state that both `branch` and `branchName` are valid,
but only `branchName` appears to work.

gaudenz committed:
Cloudscale volume (#49887)

* Cloudscale integration test setup

CloudProvider and CloudEnvironment classes for Cloudscale integration
tests. This also contains a cloudscale_common role with common
variables for all tests.

* cloudscale_volume module

New cloud module to manage volumes on the cloudscale.ch IaaS service.

vpiserchia committed:
Add Devuan Support for ascii release - resolve 49615 (#49616)

* Add Devuan Support for ascii release - resolve 49615

* Devuan Ascii does not have the VERSION_ID in the /etc/os-release file, we need to set NA

mattclay committed:
Use relative inventory paths instead of symlinks.

ganeshrn committed:
Fix issue in junos and bigip action plugins (#50636)

*  tmp is not required to be passed to in run()
   within action plugin

NilashishC committed:
New Module: vyos_ping (#50193)

* New module for vyos ping functionality

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

* Add some new parameters

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

* Need spaces

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

* Fix pylint errors

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

* Fix pylint errors

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

* Fix docs

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

* Fix docs

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

* Fix docs

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

nijinashok committed:
ovirt_storage_connection: Fix issue in detaching the connection (#50005)

Currently deleting the storage connection just tries to delete the
connection but it doesn't detach the connection from storage domain.
The patch first tries to detach the connection from storage domain
before attempting to delete the same if the storage domain parameter
is provided.

trishnaguha committed:
fix eos_l2_interface invalid command (#50644)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

felixfontein committed:
ufw: extend integration tests (#50400)

* Improve cleanup.

* Add check mode tests.

Failing tests commented out; will be fixed in #49948.

* Add reload and reset tests.

* Add tests for other global state.

* Work around ufw bugs.

WojciechowskiPiotr committed:
docker_node_facts: Read information about swarm node from swarm manager (#50571)

* docker_node_facts: New module to read information about swarm node from swarm manager

* Update lib/ansible/modules/cloud/docker/docker_node_facts.py

Co-Authored-By: WojciechowskiPiotr <23406016+WojciechowskiPiotr@users.noreply.github.com>

* Update lib/ansible/modules/cloud/docker/docker_node_facts.py

Co-Authored-By: WojciechowskiPiotr <23406016+WojciechowskiPiotr@users.noreply.github.com>

* Update lib/ansible/modules/cloud/docker/docker_node_facts.py

Co-Authored-By: WojciechowskiPiotr <23406016+WojciechowskiPiotr@users.noreply.github.com>

* docker_node_facts: min_docker_api_version and docker updated to align with Docker documentation

* docker_node_facts: Updated documentation and minimum version requirements

* docker_node_facts: Updated documentation of requirements

* Update lib/ansible/modules/cloud/docker/docker_node_facts.py

Co-Authored-By: WojciechowskiPiotr <23406016+WojciechowskiPiotr@users.noreply.github.com>

* Update lib/ansible/modules/cloud/docker/docker_node_facts.py

Co-Authored-By: WojciechowskiPiotr <23406016+WojciechowskiPiotr@users.noreply.github.com>

* Update lib/ansible/modules/cloud/docker/docker_node_facts.py

Co-Authored-By: WojciechowskiPiotr <23406016+WojciechowskiPiotr@users.noreply.github.com>

* Update lib/ansible/modules/cloud/docker/docker_node_facts.py

Co-Authored-By: WojciechowskiPiotr <23406016+WojciechowskiPiotr@users.noreply.github.com>

* Update lib/ansible/modules/cloud/docker/docker_node_facts.py

* Update lib/ansible/modules/cloud/docker/docker_node_facts.py

Co-Authored-By: WojciechowskiPiotr <23406016+WojciechowskiPiotr@users.noreply.github.com>

locutus304 committed:
fix ip address tokenization bug in eos_static_route module (#43151)

rwinand committed:
Introduce 'insertbefore' and 'insertafter' to specify the position (#44811)

* Introduce 'insertbefore' and 'insertafter' to specify the position children have to be inserted.

* Added version_added to new options

* Xpath in example needs single quotes

* Added integration tests for insertafter and insertbefore

* Changed version_added to 2.8

sivel committed:
Make sure to clean up even if an exception occurs (#50620)

richlv committed:
Update zabbix_action.py (#50582)

* Document "host_groups", "templates" and "inventory" operation parameters.
* Fix a couple of typos.

Fixes #50581

+label: docsite_pr

bcoca committed:
avoid x2 setting of set_fact when 'cacheable' (#50564)

* avoid x2 setting of set_fact when 'cacheable'

  fixes #50556

* ammend docs to new behaviour

* added period

sivel committed:
Perf graphing (#46346)

* csv of memory usage

* Fix var

* Configurable output file

* Add cpu profiling

* Valdiate the existence of cgroup files

* Add guard to prevent exception when trying to reset max memory value

* to_bytes/to_text and docs updates

* Add support for CPU results

* Just track the max, don't log all results, and then calculate max

* Restore cgroup_memory_recap, and move new functionality into cgroup_perf_recap

* Add pid count tracking, restructure to support more profilers

* Add cli tool for graphing cgroup_perf_recap data

* csv_output_dir is a path

* Correct CALLBACK_NAME

* Include uuid in csv data

* fix linting errors

* Bump version_added

* Create helper funciton to create dict from list of keys, with callable default

* Updated notes to include pids

* Print a newline after each section

* Plugin improvements

* Add option to supporess recap display
* Add default for output directory
* Add option to dictate whether or not to write files
* Add JSON-seq output option

* s/uuid/task_uuid

* Use bytes for paths

* Increase polling interval length for pids/memory

* Reduce instance attrs, change how we invoke profilers

* Shorten some line lengths

* Remove more instance attrs

* Fix some typos

* document directory creation, and catch exceptions

* Enable per task file outputs, and filename customization

* s/per_task_file/file_per_task/g

kkarolis committed:
Documentation fix for developing own plugin (#50547)

Seems like `self` is not necessary in the call args

+label: docsite_pr

mattclay committed:
Improve performance of no-tests-as-filters test.

mattclay committed:
Disable retry files for integration tests.

felixfontein committed:
docker_network IPAM test / general docker test cleanup (#50499)

* Re-enable docker_network tests.

* Basic cleanup of docker daemon.

* Add docker CLI detection.

* YAML notation.

kairoaraujo committed:
Included support to AIX group of subsystems on service module (#33043)

* Included support to AIX group subsystems

AIX systems has subsystems as services but also uses group
subsystems.
For example, spooler is a group subsystem to services
qdaemon, writesrv, and lpd.
This change enables the possibility to use also the group
susbsystmes such as spooler, nfs, etc.

When the service name is informed, first the module will
check if the name is a subsystem, if not it will check if
the name is a group subsystem and also it subsystems states.

This change makes services more flexible with AIX systems.

* Included test/legacy/aix_services.yml for tests

As discussed on IRC ansible-devel channes, was include the
legacy tests for further manual tests.

gforster committed:
Add AIX support to reboot module (#50353)

* Add ability for reboot module to work for AIX

* changelog for AIX reboot support

dagwieers committed:
team_aix: Add members to the new team (#50660)

orgito committed:
Add auth_timeout parameter when supported by paramiko (#50448)

* Add auth_timeout parameter when supported

Paramiko 2.2 introduces the auth_timeout parameter. This will set the
parameter to the same value of the timeout parameter to prevent
"Authentication timeout" errors.

* Conditionally add auth_timeout to ssh.connect

Renamed sock_kwarg to ssh_connect_kwargs and conditionally added the
auth_timeout parameter based on the installed paramiko version.

* Add changelog fragment

jborean93 committed:
win_psexec: added the pid return value to the docs (#50664)

rambleraptor committed:
New module: gcp_bigquery_table_facts (#50679)

rambleraptor committed:
New module: gcp_bigquery_dataset_facts (#50677)

jborean93 committed:
adbotmeta: add windows support to executor/powershell i (#50689)

triplepoint committed:
inventory: vagrant: rename deprectated ansible_ssh_*  (#50694)

resmo committed:
zabbix: fix author's github handle (#50700)

mindrb committed:
zabbix_host_facts: get list of templates (#50007)

Get list of templates among other items. It is useful because later one can edit this list using zabbix_host module.

rcarrillocruz committed:
Checkpoint access rule (#49937)

* WIP checkpoint_access_rule module

* Add fixes and docstrings

* Add dunder init

* Fix sanity tests issues

* Fix sanity test

* Add RETURN and EXAMPLES

* Fix example

* Fix pep8

* Add tests

* Fix pep8

* Fix pep8

dagwieers committed:
Added two new members to AIX WG

felixfontein committed:
Remove elements= from arg_spec when type is not 'list'. (#50484)

acozine committed:
fix docs for tower modules (#50710)

jainnikhil30 committed:
Fixing exception import for tower modules (#50447)

* fixing the exception import from tower modules

* Adding tests for checking tower modules are failing with correct msg

* fixed failing tests

* fixed failing test in tower_team

mkrizek committed:
tqm._initialize_notified_handlers() is noop now (#50713)

richlv committed:
Update zabbix_action.py (#50548)

* Update zabbix_action.py

Clarify when in/not in operators can be used.

Commit:
Lenovo cnos banner new module (#50145)

* Adding cnos_banner module to the list.

Commit:
Lenovo cnos l2interface (#50641)

* Adding cnos_l2 interface module and its associated Integration test as well as Unit test code

jplitza committed:
Fix sanitizing config lines (#50553)

Previously, the index got out of sync with the actual config list. Invoked with lines:
  - bad
  - first
  - bad
  - second
the sanitization would first delete index 0 and then index 2, which would result in the output
  - first
  - bad
By reversing the list, we avoid that problem (though a filter() would be nicer)

Akasurde committed:
default: callback plugin allow unreachable task to stderr (#50533)

Provide toggle flag to allow display of unreachable task to stderr
using default callback plugin.

Fixes: #48069

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

felixfontein committed:
Add delay to fix spurious test breaks.

sivel committed:
Add tests for WANT_JSON and old style modules (#50555)

* Add tests for WANT_JSON and old style modules

* quote source path

* Attempt to further appease tests

* Check for file and not just exists

* omg don't be dumb

* moar fixes

* shellcheck is the worst :)

* Test the custom modules for failure without arg files

mattclay committed:
Move any_errors_fatal test inventory into test.

mattclay committed:
Move include_import test inventory into test.

mattclay committed:
Move meta_tasks test inventory into test.

mattclay committed:
Move strategy_linear test inventory into test.

dagwieers committed:
Add mksysb module to $team_aix

Commit:
cleanup - required: false (#50648)

mattclay committed:
Merge pull request #50735 from mattclay/callback-inv

Move callback_default test inventory into test.

yuwzho committed:
azure_managed_disk enhancement (#50317)

justjais committed:
To resolve NIOS race condition when ip allocated via NIOS next available ip function (#49818)

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* resolves bug 45218

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* fixing review comments

justjais committed:
PR to fix the issue of keyerror in Nios api (#49183)

* ipv4addr keyerror fix

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* fix review comment

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

rajaspachipulusu17 committed:
Pluribus Networks pn_dscp_map_pri_map module with Unit tests (#50142)

* Pluribus Networks pn_dscp_map_pri_map module with Unit tests
* Corrected method name in unit test script

yuwzho committed:
Curate the  virtual network facts module (#50311)

ikemo3 committed:
Fix: lxca_nodes misspelled in the lxca_common.py (#50734)

misspelled "requirements" to "requirement".

https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html

tetsuyasodo committed:
fix eos_l2_interface insufficient commands (#50754)

anasbouzid committed:
schema can also be provided if type is default_privs (#50498)

+label: docsite_pr

Commit:
reviewed module using CSharpUtil Ansible.Basic

Commit:
Merge branch 'win_mssql_query' of https://github.com/dlazz/ansible into win_mssql_query

dlazz commented:
Closing this because I made a mess with rebase. I will open antother one
